### ISSUE TITLE
AudioStreamPlayer: play() resets pitch_scale back to 1 in Web Export

### ISSUE THREAD

**Issue by Ivorforce** [godotengine/godot#95850]
### Tested versions

- Reproducible in 4.3

### System information

Godot v4.3.stable - macOS 13.6.7 - Vulkan (Forward+) - dedicated AMD Radeon RX 6600 XT - 11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz (16 Threads)

### Issue description

In the web export, setting the pitch scale, then playing a sound resets the pitch scale. This was described in #53585 and fixed in #53641. However, the problem persists for me in the web export.

### Steps to reproduce

If you play a sound, then set the pitch scale, it plays it correctly:
```gdscript
	talk_sound.play()
	talk_sound.pitch_scale = randf_range(0.15, 0.23)
```

If you set a pitch scale, then play a sound, it plays it at pitch scale 1 (incorrect):

```gdscript
	talk_sound.pitch_scale = randf_range(0.15, 0.23)
	talk_sound.play()
```

### Minimal reproduction project (MRP)

None yet. I'm a bit busy with a game jam to test it with a minimal reproducer, so if you're busy, wait a few days and I'll test it myself with the reproducer from #53585. It _is_ reproducible in https://github.com/Ivorforce/Built-To-Scale though - specifically https://github.com/Ivorforce/Built-To-Scale/blob/main/scripts/conversation/conversation_box.gd, if adjusted to the faulty setup. The game comes to a dialogue box within a few seconds of starting.

[95850.zip](https://github.com/user-attachments/files/16677122/95850.zip)

_Production edit: added MRP_

**Comment by akien-mga** [comment]
CC @adamscott

**Comment by BaconEggsRL** [comment]
Hi, just reporting I also noticed this issue in 4.3
I'm not sure how but it seemed like it fixed itself after I started trying to print out the pitch_scale before and after the play() operation

**Comment by adamscott** [comment]
> Hi, just reporting I also noticed this issue in 4.3 I'm not sure how but it seemed like it fixed itself after I started trying to print out the pitch_scale before and after the play() operation

The fix was cherry-picked for 4.3.1, not yet released.
https://github.com/godotengine/godot/pull/95862#issuecomment-2353212791
